### DISCUSSION TITLE
C++ Client Libraries for Dialogflow V2_beta1

### DISCUSSION THREAD

**Discussion by PratheepCXM** [googleapis/google-cloud-cpp#9426] [Q&A] [answered]
By Default, the GITHUB Source creates C++ Client libraries for Dialogflow V2. The repository contains the proto files for Dialogflow V2_Beta1. How to create the C++ client libraries  for v2beta1 using the existing CMAKE Scripts? .

**Comment by coryan** [comment]
> How to create the C++ client libraries for v2beta1 using the existing CMAKE Scripts?

At the moment there is no way to do this.

**Reply by PratheepCXM** [reply]
Does it mean the application must use the GRPC API, using the protos available in the v2beta1 directory of Dialogflow? 
Whether any CMake scripts available alone to create the grpc.pb.h ,grpc.pb.cpp and Dialogflowprotos.lib from V2beta1 Protos to call GRPC API directly from an application?

**Reply by coryan** [reply]
> Does it mean the application must use the GRPC API, using the protos available in the v2beta1 directory of Dialogflow?

At this time, yes.  In general, we only create C++ libraries for GA products and only for the GA version of each product's API.  Let me check what is the status of this API.  Maybe despite its name, maybe the `v2beta1` API is "GA".  That would make me very sad, but it could be the case.

> Whether any CMake scripts available alone to create the grpc.pb.h ,grpc.pb.cpp and Dialogflowprotos.lib from V2beta1 Protos to call GRPC API directly from an application?

We only create the CMake scripts for specific protos when we create the library for them.  The procedure is [fairly involved](https://github.com/googleapis/google-cloud-cpp/blob/main/doc/contributor/howto-guide-adding-generated-libraries.md), though easy to follow if you have seen it done.

**Reply by PratheepCXM** [reply]
Thanks coryan. kindly let us know when we will getV2beta1 API GA.

**Reply by coryan** [reply]
So, I just heard from the product team.  `v2beta1` is not GA, and there is no plan to make it GA.  You should be using the `v2` or the `v3` APIs.  `v3` is also known as `CX` and it has a client library:

https://github.com/googleapis/google-cloud-cpp/tree/main/google/cloud/dialogflow_cx#readme

**Reply by PratheepCXM** [reply]
Coryan,

I need to do integration with CCAI Agent Assist. V2beta1 only have the Streaming Analyze Content API. I want to use this for real time streaming the VOICE. 

Is it possible to use the CX for AgentAssist? I have used ES as the google documentation specifies that Agent assist is the Extension of ES.

**Reply by coryan** [reply]
I think you may be asking about this feature:

https://github.com/googleapis/googleapis/commit/b9927eb6fe36783981201eecdf0b95aa340c9137

as best as I can tell that feature is not GA, and not available in the `v2` or `v3` versions of the API.  I do not have a good answer for you at this time.  Give me a few days to get back to you.

**Reply by coryan** [reply, marked as answer]
FWIW, the next version of `google-cloud-cpp` will include `StreamingAnalyzeContent` as part of the v2 API.

https://github.com/googleapis/google-cloud-cpp/blob/2ae1f558bd18dde0946c0238f2d0b2301bee4f71/google/cloud/dialogflow_es/participants_client.h#L358-L361
